risc0/zkvm/src/prove/mod.rs

```diff
@@ -292,52 +292,53 @@ where
 
         let seal = prover.finalize(&[&mix, &out], eval.as_ref());
 
-        let receipt = SegmentReceipt {
+        let receipt = SegmentBaseReceipt {
             seal,
             index: segment.index,
         };
         let hal = CpuVerifyHal::<_, H::HashSuite, _>::new(&crate::CIRCUIT);
         receipt.verify_with_hal(&hal)?;
 
-        Ok(receipt)
+        Ok(Box::new(receipt))
     }
 }
 
 fn provers() -> HashMap<String, Rc<dyn Prover>> {
     let mut table: HashMap<String, Rc<dyn Prover>> = HashMap::new();
     {
-        let prover = Rc::new(LocalProver::new("cpu", cpu::sha256_hal_eval()));
-        table.insert("cpu".to_string(), prover.clone());
-        table.insert("$default".to_string(), prover);
+        let prover = Rc::new(LocalProver::new("cpu:sha256", cpu::sha256_hal_eval()));
+        table.insert("cpu:sha256".to_string(), prover.clone());
+        table.insert("$sha256".to_string(), prover);
 
         let prover = Rc::new(LocalProver::new("cpu:poseidon", cpu::poseidon_hal_eval()));
         table.insert("cpu:poseidon".to_string(), prover.clone());
-        table.insert("$poseidon".to_string(), prover);
+        table.insert("$poseidon".to_string(), prover.clone());
```

Comment:
warning: we found poseidon to be far slower on CPU than the sha256 hashing so this change will probably slow down the majority of local proving.

Reply:
Bummer! Thanks for the heads up. I saw the perf numbers and was thinking that it was something else..

Reply:
I'll make a quick fix and bring this up internally

Reply:
The related benchmark shows a significant improvement in performance, we might be reading the results wrong